### ISSUE TITLE
World Boss respawn timers not random

### ISSUE THREAD

**Issue by j-hdt** [chromiecraft/chromiecraft#3219]
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [X] 60

### Current Behaviour

It seems that the world boss respawn timers are not random but on a fixed value.

### Expected Blizzlike Behaviour

the respawn timing shouldnt be predictable exactly.
instead, there should be time windows in which they respawn after being killed

### Steps to reproduce the problem

.

### Extra Notes

Currently, the predicted respawn timer is getting abused by a group of players (guild) since the initial release of the world bosses.
Moreover, it seems there is no active bug report for this issue yet - thus, the people profiting from this bug are also not reporting the issue apparently, which subverts the basic idea of Chromiecraft being a place to contribute.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/e8d3292533c7e5d18af15d9bffabd581bacbadf9

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

**Comment by Hyperbearr** [comment]
Going to address this in three parts

**World bosses are bugged**

They were, this has been confirmed by the developers, they knew and we didn't. The earliest wbosses were spawned manually by the developers whilst they fixed the timers and we had no idea. These early bosses were consistently spawned in an 8 hour window that aligned with the active hours of one of the developers, but again we didn't find this out until after the fact. 

The current windows appear to be truly random across a 24hr clock, but are not blizzlike in that the number of days between spawns isn't blizzlike. However this is what we've derived based on watching spawns, recording kill times and working it out ourselves. We've also actively reported this bug to the developers and have been told that its a server side limitation created by the way the bosses are coded that means they can't emulate the true blizzlike behaviour. 

Whatever else you may say, the timers ARE NOT predictable. 

**An unnamed guild (its Synergy, we know its Synergy) are exploiting wbosses**

No, no we aren't. We have had near 24/7 scouting during possible wboss windows (based on our own derivations) and are the only guild actively scouting with any consistency, so ofc we get all the bosses. It doesn't mean we are colluding with the developers, abusing bugs or doing any other shady stuff, we just have the largest share of committed players on the server. 

**It hasn't been reported**

Actually it has. We aren't reporting world boss bugs publicly as they are ultimately a competitive resource and so we don't want to share the data with other guilds that we've put a lot of time and effort into procuring and analysing. 

Is that sweaty? Absolutely. 
Is it unreasonable? We don't think so. 

Just because we haven't posted it on github though doesn't mean it hasn't been reported, the devs have been made aware via discord DM's and support tickets. 

Prior to release there was a HUGELY exploitable bug available with the .npc info command that would tell you the exact world boss spawn down to the second, and we reported this before they ever went live via the discord, to the devs in DMs and we shared it with officers from other guilds we suspected might compete (Primals, Tribunal, BBBB, etc etc) to ensure that no one had an unfair advantage. This bug was resolved prior to launch thanks to our actions, if we wanted to abuse we could've done so and in a much more egregious fashion. 

Finally, just want to add that a lot of the world boss issues were aired publicly in the discord by our members too in the last week. 

TL:DR; It's reported, good on you for reporting it, but please refrain from making baseless accusations - ESPECIALLY in bug reports, it adds nothing of value.

**Comment by Spook9** [comment]
Please dont post bug-reports without evidence. This only takes time away from the devs and since it cant be confirmed this post adds no value.

**Comment by j-hdt** [comment]
https://i.imgur.com/0PrRhxc.png

> GitHub issues are the only valid way top report bugs. They are all tracked one after another. Due to the limited time we can invest in the project, reports can not be triaged instantly.
> Since the bug was reported just today and isn't even answered, we can't possibly have checked on the behaviour. If rules were broken, we will act accordingly. Thanks for bringing it up.

I will leave this here as a statement from the CC team, maybe you should reconsider how you approach bugs you are encountering and take the same approach CC is taking - communicating them transparently. That way wrong impressions can be prevent.

> Going to address this in three parts
>
> **It hasn't been reported**
> 
> Actually it has. We aren't reporting world boss bugs publicly as they are ultimately a competitive resource and so we don't want to share the data with other guilds that we've put a lot of time and effort into procuring and analysing.
> 
> Is that sweaty? Absolutely. Is it unreasonable? We don't think so.
> 
> Just because we haven't posted it on github though doesn't mean it hasn't been reported, the devs have been made aware via discord DM's and support tickets.

**Comment by Hyperbearr** [comment]
> https://i.imgur.com/0PrRhxc.png
> 
> > GitHub issues are the only valid way top report bugs. They are all tracked one after another. Due to the limited time we can invest in the project, reports can not be triaged instantly.
> > Since the bug was reported just today and isn't even answered, we can't possibly have checked on the behaviour. If rules were broken, we will act accordingly. Thanks for bringing it up.
> 
> I will leave this here as a statement from the CC team, maybe you should reconsider how you approach bugs you are encountering and take the same approach CC is taking - communicating them transparently. That way wrong impressions can be prevent.
> 
> > Going to address this in three parts
> > **It hasn't been reported**
> > Actually it has. We aren't reporting world boss bugs publicly as they are ultimately a competitive resource and so we don't want to share the data with other guilds that we've put a lot of time and effort into procuring and analysing.
> > Is that sweaty? Absolutely. Is it unreasonable? We don't think so.
> > Just because we haven't posted it on github though doesn't mean it hasn't been reported, the devs have been made aware via discord DM's and support tickets.

There are some bugs which are exploitable (server crashes, duplication bugs, rare spawns/world boss timers) which have been omitted from that blanket rule and which are allowed to be reported via tickets, and the staff have acknowledged this plenty of times in the discord, often even telling people to put those bugs into tickets when they enquire.

**Comment by temperrr** [comment]
I can confirm that the respawn time is random.
